pkg/services/authn/clients/jwt.go

```diff
@@ -143,21 +144,14 @@ func (s *JWT) Test(ctx context.Context, r *authn.Request) bool {
 		return false
 	}
 
-	// The header is Authorization and the token does not look like a JWT,
-	// this is likely an API key. Pass it on.
-	if s.cfg.JWTAuthHeaderName == "Authorization" && !looksLikeJWT(jwtToken) {
+	// If the "sub" claim is missing or empty then pass the control to the next handler
+	if !authJWT.HasSubClaim(jwtToken) {
```

Comment:
opinionated nit: I'd merge the last 2 conditional statements into a single return value. Something similar to this.
```
func (s *JWT) Test(ctx context.Context, r *authn.Request) bool {
	if !s.cfg.JWTAuthEnabled || s.cfg.JWTAuthHeaderName == "" {
		return false
	}

	jwtToken := s.retrieveToken(r.HTTPRequest)

    // If the "sub" claim is missing or empty then pass the control to the next handler
    return jwtToken != "" || authJWT.HasSubClaim(jwtToken);
}
```

Reply:
Thanks for the suggestion! I checked the cognitive complexity and the cyclomatic complexity of the current implementation and your suggestion and those are the same, so I wouldn't change this part, because it is easier and quicker (at least for me) to understand the pre-checks for this auth client.